projects/auth0-angular/src/lib/auth.service.spec.ts

```diff
@@ -136,6 +137,34 @@ describe('AuthService', () => {
     });
   });
 
+  describe('The `idTokenClaims` observable', () => {
+    it('should get the ID token claims if authenticated', (done) => {
+      const claims: IdToken = {
+        __raw: 'idToken',
+        exp: 1602887231,
+        iat: 1602883631,
+        iss: 'https://example.eu.auth0.com/',
+      };
+
+      (auth0Client.isAuthenticated as jasmine.Spy).and.resolveTo(true);
+      (auth0Client.getIdTokenClaims as jasmine.Spy).and.resolveTo(claims);
+
+      service.idTokenClaims$.subscribe((value) => {
+        expect(value).toBe(claims);
+        done();
+      });
+    });
+
+    it('should get the ID token claims if not authenticated', (done) => {
+      (auth0Client.isAuthenticated as jasmine.Spy).and.resolveTo(true);
```

Comment:
I know the existing tests are identical (like for getUser), but why does the test mention `if not authenticated` but does use `resolveTo(true)`?

Reply:
Good spot, something doesn't seem right here. If a user wasn't authenticated I would not expect to be able to get the claims at all. That observable should not emit￼ anything.

Reply:
I was wondering this also. I think the test should read `should not get the ID token claims if not authenticated` and `resolveTo(false)`.

Reply:
It might also be trying to test that if the user is authenticated, but there are no claims, it still emits something being falsey.

Reply:
I've updated the negative tests for user$ and idTokenClaims$ in latest commit